File: app/src/main/java/com/abishov/hexocat/home/HomeBindings.java
package com.abishov.hexocat.home;

import com.abishov.hexocat.commons.dagger.FragmentScope;
import com.abishov.hexocat.home.trending.TrendingFragment;
import com.abishov.hexocat.home.trending.TrendingModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class HomeBindings {

    @FragmentScope
    @ContributesAndroidInjector(modules = TrendingModule.class)
    abstract TrendingFragment contributesTrendingFragmentInjector();
}

File: app/src/main/java/com/abishov/hexocat/home/HomeBindings.kt
package com.abishov.hexocat.home

import com.abishov.hexocat.common.dagger.FragmentScope
import com.abishov.hexocat.home.trending.TrendingFragment
import com.abishov.hexocat.home.trending.TrendingModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class HomeBindings {

  @FragmentScope
  @ContributesAndroidInjector(modules = arrayOf(TrendingModule::class))
  internal abstract fun contributesTrendingFragmentInjector(): TrendingFragment
}
